packages/patternfly-4/react-core/src/components/Table/Header.js

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import { Header } from 'reactabular-table';
+import PropTypes from 'prop-types';
+import { TableContext } from './Table';
+import { isEqual } from 'lodash';
+
+const propTypes = {
+  className: PropTypes.string,
+  headerRows: PropTypes.array
+};
+
+const defaultProps = {
+  className: ''
+};
+
+class ContextHeader extends React.Component {
+  shouldComponentUpdate(nextProps) {
```

Comment:
I'd like to discuss this component more deeply because it requires lodash as dep and that might be unwanted/not necessary. What I wanted to achieve with this is that I don't like passing columns to table provider + reactabular is using outdated `getChildContext` and I'd like to avoid it by using our context. But that requires passing columns from child element to parent element and that might be a bit confusing because we are basicaly changing parent data from child.

Reply:
IsEqual is a deep compare. It is likely that this is actually sloeer than letting react run the renders and the vdom diffing. I haven't looked at anymore of this component so maybe it is needed in some way, but the lodash equal will likely need to change

Reply:
That's another reason why I am trying to bring conversation about this thing. The problem is that every time header is changed (new column, d&d, etc.) we have to send updated data to parent trough `Context`. If that happens parrent will try to rerender all of it's children and one of these is actually this component so we end up in never ending cycle or renders.

So there are multiple options... do not use columns on `header` and use them from provider (kinda strange so that's the reason I tried avoiding this).
Keep this deep compare here, honestly I don't think that the header will be changed multiple times (sorting will not effect it) so I don't think that this will be an issue.

Reply:
It might be better to start with the desired API for the table. That will likely solve most of this. I do like the idea of providing our own. What is we did something like this:

In React Core provide the styled primitives
```
import { Table, TableCell, TableRow, TableHead, TableBody } from '@patternfly/react-core';
```
Those would be strictly dumb styled elements.

Then we could offer a more opinionated table from a package `@patternfly/react-tabular` or that follows react-tabular's api but uses our components, and adds some other sane overrides. 

This approach would allow a user to either build their own tables or use our opinionated sorting solution. Sorting is not that hard So personally pulling in a lib for this doesn't seem optimal, but I could see if helping people with more complex solutions.

DIY Table Example
```tsx

import { Table, TableCell, TableRow, TableHead, TableBody } from '@patternfly/react-core';

class DataTable extends React.Component {
private handleSortDate = (sortBy: keyof ExampleData) => {....}

public render() {
  return (
      <Table>
        <TableHead>
          <TableRow>
            <TableCell sortDirction={sortBy === 'name' ? sortDir : null} onClick={() => this.sortData('name')}>Name</TableCell>
          </TableRow>
        <TableHead>
        <TableBody>
          {data.map((data) => (
            <TableRow key={data.id}>
              <TableCell label="name">{data.name}</TableCell>
            </TableRow>
           )})}
        </TableBody>
      </Table>
  );
}
```